scrnaseq_hto.Rmd

```diff
@@ -220,34 +215,51 @@ hto_levels = c("Negative", "Doublet", hto_samples[1:length(hto_samples)])
 names(hto_colours$col_hto_collapsed) = hto_levels
 ```
 
-```{r hto_cells_summary}
-summary = purrr::map_dfr(original_assay_names, function(a) {
-  data.frame(Assay=a,
-             CellsTotal=length(assay_cells_hto_zero),
-             CellsWithHtoReads=sum(!assay_cells_hto_zero),
-             CellsWithoutHtoReads=sum(assay_cells_hto_zero))
-  })
+## HTO counts
+Sometimes demultiplexing fails, because some HTOs generally do not perform well, or some cells end up with zero HTO counts. To assure the script can run through, we remove HTOs and cells with zero counts. 
+```{r hto_counts, results="asis"}
+# Check for HTOs with 0 HTO counts, and remove them if necessary 
+hto_counts = Seurat::GetAssayData(sc, assay="HTO", slot="counts") 
+hto_counts_row = hto_counts %>% Matrix::rowSums() %>% as.data.frame()
+colnames(hto_counts_row) = "HtoCounts"
+if (any(hto_counts_row$HtoCounts == 0)) {
+  sc = subset(sc, features=rownames(which(hto_counts_row$HtoCounts == 0)))
```

Comment:
The subset here does not work I would say:

Try this:

`sc[["HTO"]] = subset(sc[["HTO"]], features=rownames(which(hto_counts_row$HtoCounts > 0)))`